### PR TITLE
Add Workspaces in Release pipeline instead of PipelineResources

### LINKED ISSUE #1019: Use workspaces instead of PipelineResource on release pipeline
We are using PipelineResources and instead let's move the release pipeline to
use workspaces and git-task.yaml

/kind cleanup

### PR DESCRIPTION
**Pull request by vinamra28**
# Changes

Currently release pipeline is using PipelineResources so instead of that we are replacing them with workspaces and also adding the git-clone task.

Closes #1019 

Signed-off-by: vinamra28 <vinjain@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by vinamra28** [comment]
@vdemeester could you please review :sweat_smile:

File: tekton/release-pipeline-run.yml [hunk 1]
```diff
@@ -1,24 +1,28 @@
 ---
-apiVersion: tekton.dev/v1alpha1
-kind: PipelineResource
+apiVersion: v1
+kind: PersistentVolumeClaim
```
Comment (vdemeester) [inline tekton/release-pipeline-run.yml:3]: Mid-term, we should use `volumeClaimTemplate` for the workspace (and thus not having to declare this PVC)

**Pull request merged**